### PR TITLE
Add support for creating Zip and NuGet packages

### PR DESCRIPTION
**Pull request by slewis74**
Adds builders for Zip and NuGet packages. Config parameters are based off the existing list from the C# CLI

### CODE REVIEW

File: src/features/packages/pack/nugetPackageBuilder.test.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import fs, { rmdir } from "fs";
+import os from "os";
+import path from "path";
+import AdmZip from "adm-zip";
+import { Logger } from "../../../logger";
+
+describe("Can create a NuGet packages", () => {
+    const logger: Logger = {
+        debug: (m) => console.log(m),
+        info: (m) => console.log(m),
```
Comment (jbristowe) [inline src/features/packages/pack/nugetPackageBuilder.test.ts:11]: No need for `warn`?

File: src/features/packages/pack/zipUtils.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+export async function doZip(
+    inputFilePatterns: string[],
+    outputFolder: string,
+    zipFilename: string,
+    logger: Logger,
+    compressionLevel?: number,
+    overwrite?: boolean
+): Promise<void> {
+    const archivePath = path.resolve(outputFolder, zipFilename);
+    logger.info?.(`Writing to package: ${archivePath}`);
```
Comment (jbristowe) [inline src/features/packages/pack/zipUtils.ts:18]:
```suggestion
    logger.info?.(`Writing to package: ${archivePath}...`);

```

File: src/features/packages/pack/zipUtils.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    const zip = new AdmZip();
+
+    const files = await expandGlobs(inputFilePatterns);
+    for (const file of files) {
+        logger.debug?.(`Adding file: ${file}`);
```
Comment (jbristowe) [inline src/features/packages/pack/zipUtils.ts:23]:
```suggestion
        logger.debug?.(`Adding file: ${file}...`);

```

File: src/features/packages/pack/nugetPackageBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+export class NuGetPackageBuilder {
+    async pack(args: NuGetPackArgs): Promise<string> {
+        const archiveFilename = `${args.packageId}.${args.version}.nupkg`;
+        const tmpFolder = os.tmpdir();
+        const inputFilePatterns = args.inputFilePatterns;
+
+        if (args.nuspecArgs) {
+            const nuspecFile = path.join(tmpFolder, `${args.packageId}.nuspec`);
+            fs.writeFileSync(nuspecFile, '<?xml version="1.0" encoding="utf-8"?>\n');
+            fs.appendFileSync(nuspecFile, '<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">\n');
```
Comment (jbristowe) [inline src/features/packages/pack/nugetPackageBuilder.ts:26]: Is this the latest version? Are there other versions we should consider?

**Review by jbristowe — APPROVED** [review]

File: src/features/packages/pack/nugetPackageBuilder.test.ts (hunk 1 above)
Comment (slewis74) [reply src/features/packages/pack/nugetPackageBuilder.test.ts:11]: Not for this test, the code only writes debug and info messages.

File: src/features/packages/pack/nugetPackageBuilder.ts (hunk 4 above)
Comment (slewis74) [reply src/features/packages/pack/nugetPackageBuilder.ts:26]: I got this from https://learn.microsoft.com/en-us/nuget/reference/nuspec#general-form-and-schema, so pretty sure it's latest.

**Pull request merged**